SalesforceFoundation/EDA
# [EDA] Extend TDTM to allow disabling via User Profile

# Critical Changes
## Disable Trigger Handlers for Specific User Profiles
We've introduced a new feature to disable TDTM trigger handlers with user profiles. The "Profiles to Exclude" field is automatically added to Trigger Handler in new orgs, but in existing orgs, we recommend that you add the new field to your page layout. See [Display a New Custom Object After a Product Release](https://powerofus.force.com/s/article/EDA-Display-Cust-Obj-Post-Rel).

To access this feature, from App Launcher, find and select Trigger Handler. Select a Trigger Handler to edit and enter one or more user profiles, separating each with a semicolon, in the Profiles to Exclude field. For example, "System Administrator;Marketing User."

If you've created translated versions of user profile names in Translation Workbench, be sure to include the translated version in the Profiles to Exclude field. For a Spanish translation, for example, specify "System Administrator;Administrador del sistema."

See [Disable Trigger Handlers for Specific Users](https://powerofus.force.com/s/article/EDA-Disable-Trigger-Handlers#ariaid-title3).

# Changes

# Issues Closed
Closes #638 

# New Metadata
Profiles_to_Exclude__c
TDTM_ExcludedProfiles
TDTM_ExcludedUserNames

# Deleted Metadata

# Testing Notes
The feature can be tested on any TDTM, below is an example:
1) Goto EDA -> Trigger Handlers -> Select the record for 'TERM_CannotDelete_TDTM'
2) Make sure Set up -> Custom Settings -> Hierarchy Settings -> 'Term Cannot Delete' is set to True 
3) Create a Term with at least one Term Grade child record
4) Delete the record. The record deletion is prevented since it has child records.
5) Add 'System Administrator' (with no quotes) into the field 'Profiles to Exclude'. If you want to add multiple profiles use semi colon Ex: 'System Administrator; Standard User'.
6) Make sure you're system administrator -> Set up -> User -> select your user -> Profile -> 'System Administrator'
6) Now try deleting the Term record (Same as step 4)
7) The term should be deleted

Testing Translations:
1) Repeat Step 1 - 4 from above
2) Add 'Administrador del sistema' to Profiles to Exclude
3) Change your language to Spanish. Set up -> User -> select your user -> language -> Spanish
4) Now try deleting the Term record 
5) The term should be deleted
